.github/workflows/cigh.yml

```diff
@@ -0,0 +1,124 @@
+# This workflow aims at building and testing FreeCAD
+# It uses a runner based on latest Ubuntu LTS version
+# It builds using CMake/CCache+Make/Clang as a toolchain
+#
+# It has been foreseen to be used with CMake/Ninja/Clang...
+# ...but this is disabled at the moment because Ninja...
+# ...only works based on file timestamps, which isn't...
+# ...compatible with Git not managing them
+
+name: FCBuildNTest
+
+on:
+  workflow_dispatch: # Triggered manually
+# To run manually, use :
+# curl -X POST -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/:OWNER/FreeCAD/actions/workflows/cigh.yml/dispatches -d '{"ref":":BRANCH_NAME"}' -u :USER_NAME
+# It will then require your Github PAT to trigger the launching event
+    branches-ignore: # Do not run on master and release branches
```

Comment:
from what I understood this section deals with manually triggered runs, does it make sense to ignore branches here? I see two possibilities here: 
-the ignore config is ironically ignored when doing a manual run in which case these lines are meaningless
-the ignore config is respected which disables manually running the CI on master and release branches, I don't think this is desired, presumably if someone is trying to run it on one of those branches it is for a reason. I recall instances were maintainers wanted the CI to run on a release branch before publishing binaries.

As always my knowledge of these things is limited so please take my comments with a grain of salt.

Reply:
As said, this are guardrails, essentially ATM to prevent interfering with gitlab CI.